File: MyUdemyLearningApp/app/src/main/java/com/codewithamit/myudemylearningapp/MainActivity.java
package com.codewithamit.myudemylearningapp;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }


    public void converdollertorupeebutton(View view) {

        EditText number=(EditText) findViewById(R.id.editTextNumber);
        Double num = Double.parseDouble(number.getText().toString());
        Double rup=num*74.5;
        Toast.makeText(MainActivity.this,"Rs: "+rup,Toast.LENGTH_SHORT).show();

        TextView rupTextview = (TextView) findViewById(R.id.textView3);
        rupTextview.setText((Double.toString(rup)));
    }
}

File: androidRoom/app/src/main/java/com/example/androidroom/MainActivity.kt
package com.example.androidroom

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.room.Room
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import java.util.Observer

class MainActivity : AppCompatActivity() {

    val db by lazy {
        Room.databaseBuilder(
            this,
            AppDatabase::class.java,
            "User.db"
        )
            .fallbackToDestructiveMigration()
            .build()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnsave.setOnClickListener {
            GlobalScope.launch(Dispatchers.IO) {
                db.UserDao().inster(User("amit", "7875709937", "goa", 20))
            }
        }


        db.UserDao().getAllUser().observe(this, androidx.lifecycle.Observer { list ->

            if (list.isNotEmpty()) {
                with(list[list.size - 1]) {
                    tvname.text = name + id
                    tvadr.text = address
                    tvnumber.text = number
                    tvage.text = age.toString()
                }
            }

        })
    }

}
